soc/wch/ch32v00x/vector.S

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024 Michael Hope
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/toolchain.h>
+
+/* Exports */
+GTEXT(__start)
+
+/* Imports */
+GTEXT(__initialize)
+
+SECTION_FUNC(vectors, ivt)
+	.option norvc
+	j       __start
+	.rept 38
+	.word	_isr_wrapper
```

Comment:
Do you really wrap all traps together in zephyr?

Reply:
Yes, this is Zephyr's approach. There's a common RISC-V ISR wrapper that switches to the IRQ stack, loads the ISR argument, and jumps to the ISR.